### PR TITLE
links: add link factory for record and files

### PR DESCRIPTION
**Pull request by Glignos**
References https://github.com/inveniosoftware/cookiecutter-invenio-instance/issues/172

### CODE REVIEW

File: invenio_records_files/config.py [hunk 1]
```diff
@@ -28,10 +28,11 @@
   latest/configuration.html#invenio_records_rest.config
   .RECORDS_REST_ENDPOINTS>`_ you have defined in your application.
 
-* ``<record_pid_type>`` is the PID type associated to the record defined in
-  your `Invenio-Records-REST <https://invenio-records-rest.readthedocs.io/en/
+* ``<endpoint_name>`` is the unique name of your endpoint as you can see in
+ `Invenio-Records-REST <https://invenio-records-rest.readthedocs.io/en/
   latest/configuration.html#invenio_records_rest.
-  config.RECORDS_REST_ENDPOINTS>`_ like configuration.
+  config.RECORDS_REST_ENDPOINTS>`_ like configuration.This needs to match
```
Comment (equadon) [inline invenio_records_files/config.py]:
```suggestion
  config.RECORDS_REST_ENDPOINTS>`_ like configuration. This needs to match
```

File: invenio_records_files/links.py [hunk 2]
```diff
@@ -23,3 +24,21 @@ def default_bucket_link_factory(pid):
                        bucket_id=bucket.id, _external=True)
     except AttributeError:
         return None
+
+
+def default_record_files_links_factory(pid, record=None, **kwargs):
+    """Factory for record links generation.
```
Comment (equadon) [inline invenio_records_files/links.py]:
```suggestion
    """Factory for record files links generation.
```
Or similar?

File: invenio_records_files/serializer.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        """Dump links."""
+        pid_value = request.view_args['pid_value'].value
+        url_path = '.{0}'.format(self.context.get('view_name'))
+        return {
+            'self': url_for(url_path, pid_value=pid_value, _external=True),
+            'versions': url_for(
+                url_path, pid_value=pid_value, _external=True) + '?versions',
+            'uploads': url_for(
+                url_path, pid_value=pid_value, _external=True) + '?uploads',
+        }
```
Comment (equadon) [inline invenio_records_files/serializer.py]: Could perform the `url_for` once and reuse that computed value for the 3 links.

File: invenio_records_files/config.py [hunk 4]
```diff
@@ -24,14 +24,17 @@
...
-  latest/configuration.html#invenio_records_rest.
-  config.RECORDS_REST_ENDPOINTS>`_ like configuration.
+  configurations names
+  <https://invenio-records-rest.readthedocs.io/en/latest/configuration.html
+  #invenio_records_rest.config.RECORDS_REST_ENDPOINTS>`_
+  that you have defined in your application.
+
+* ``<endpoint-prefix>`` is the unique name of the endpoint configuration as it
+  is defined in `Invenio-Records-REST
+  <https://invenio-records-rest.readthedocs.io/en/latest/configuration.html
```
Comment (kprzerwa) [inline invenio_records_files/config.py]: do you think it is worth to mention that the `endpoint_prefix` has to match `pid_type`?

**Pull request merged**